### PR TITLE
Wait for secrets to become managed by K8ssandra controller

### PR DESCRIPTION
**Pull request by adutra**
**What this PR does**:
Wait for secrets to become managed by K8ssandra controller

**Which issue(s) this PR fixes**:
None

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: test/e2e/suite_test.go [hunk 1]
```diff
@@ -839,18 +839,28 @@ func checkNodeToolStatusUN(
...
 
 // FIXME there is some overlap with E2eFramework.RetrieveDatabaseCredentials()
 func retrieveCredentials(t *testing.T, f *framework.E2eFramework, ctx context.Context, secretKey framework.ClusterKey) (string, string) {
-	secret := getSecret(t, f, ctx, secretKey)
+	secret := checkSecretExists(t, f, ctx, secretKey)
 	return string(secret.Data["username"]), string(secret.Data["password"])
 }
 
-func getSecret(t *testing.T, f *framework.E2eFramework, ctx context.Context, secretKey framework.ClusterKey) *corev1.Secret {
+func checkSecretExists(t *testing.T, f *framework.E2eFramework, ctx context.Context, secretKey framework.ClusterKey) *corev1.Secret {
```
Comment (burmanm) [inline test/e2e/suite_test.go:863]: Why was this renamed to an existing functionName? Especially since it seems to be used like "getSecret" unlike "checkSecretExists" (which I would expect to return a bool)

Comment (adutra) [reply test/e2e/suite_test.go:863]: Because other functions that wait on something with `assert.Eventually` are all called `checkSomething`.

Comment (burmanm) [reply test/e2e/suite_test.go:863]: But they don't return anything (some of them are named verifySomething also). This one is clearly used to return something instead of checking it (I might even say the eventually is in the wrong place there)

Comment (jsanda) [reply test/e2e/suite_test.go:863]: We definitely need to settle on and document naming conventions for verification functions. Can we do that separately, especially since this PR may fix some of the test flakes?

**Comment by jsanda** [comment]
@adutra what is the status on this?

**Comment by adutra** [comment]
@jsanda current status is "forgotten".

I will try to rebase and then I will revert the function name change to make @burmanm happy.

Stay tuned!

**Comment by adutra** [comment]
Status update: this is not relevant anymore. We decided not to check if secrets exist in e2e tests, so all the parts affected by this change were deleted since. I'm closing this.
